app/seller/seller-form.tsx

```diff
@@ -39,14 +47,51 @@ function SellerFormPage() {
 
     const items = useWatch({ control, name: "items" });
 
-    function onSubmit(data: PostFormType) {
-        console.log("Form Data: ", data);
+    async function onSubmit(data: PostFormType) {
+        setLoading(true);
+
+        let post: PostRequestType = {
+            caption: data.caption,
+            tags: data.tags,
+            items: [],
+        };
+
+        for (let i = 0; i < data.items.length; i++) {
+            let item = {
+                ...data.items[i],
+            };
+
+            try {
+                item.imageUrl = await uploadImage(item.imageUrl);
+                post.items.push(item);
+            } catch (error) {
+                console.error("Upload failed or returned bad URL:", error);
+                setError(`items.${i}.imageUrl`, {
+                    type: "manual",
+                    message: "Image upload failed",
+                });
+                setLoading(false);
+                return;
+            }
+        }
+
+        try {
+            const result = await createPost(post);
+            console.log("Post Success:", result);
+        } catch (error) {
+            console.error("Post Failed:", error);
+        } finally {
+            setLoading(false);
+        }
+
+        Alert.alert("Success", "Your post was created!");
+        reset();
```

Comment:
transfer to inside the try, kay maybe there was an error but we still show this alert, we also justwant to reset if na upload talaga

```tsx
try {
    const result = await createPost(post);
    Alert.alert("Success", "Your post was created!");
    reset();
} ....
````